go/fn/object_test.go

```diff
@@ -10,6 +10,106 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+func TestIsGVK(t *testing.T) {
+	input := []byte(`
+apiVersion: apps/v3
+kind: StatefulSet
+metadata:
+  name: my-config
+spec:
+  volumeClaimTemplates:
+    - metadata:
+        labels:
+          testkey: testvalue
+`)
+
+	inputNoGroup := []byte(`
+apiVersion: v3
+kind: StatefulSet
+metadata:
+  name: my-config
+spec:
+  volumeClaimTemplates:
+    - metadata:
+        labels:
+          testkey: testvalue
+`)
+
+	inputDifferentKind := []byte(`
+apiVersion: apps/v1
+kind: Service
+metadata:
+  name: my-config
+spec:
+  volumeClaimTemplates:
+    - metadata:
+        labels:
+          testkey: testvalue
+`)
+	resource, _ := ParseKubeObject(input)
+	resourceNoGroup, _ := ParseKubeObject(inputNoGroup)
+	resourceDiffKind, _ := ParseKubeObject(inputDifferentKind)
+	testCases := map[string]struct {
+		resource         *KubeObject
+		resourceNoGroup  *KubeObject
+		resourceDiffKind *KubeObject
+		group            string
+		version          string
+		kind             string
+		result           bool
+		resultNoGroup    bool
+		resultDiffKind   bool
```

Comment:
nit: usually in tests, we call the result value `expected`. To align with this, I think it would be better to name these `expected`, `expectedNoGroup`, and `expectedDiffKind`. 

Please put a comment above each one to give a description of what it is. Example: `// this is the expected result when the resource is missing the Group field`)